### DISCUSSION TITLE
Network issues with Cilium with `tunnel: geneve`

### DISCUSSION THREAD

**Discussion by hariprasadiit** [mysticaltech/terraform-hcloud-kube-hetzner#306] [Q&A] [answered]
Hello,

I've installed cluster with cilium CNI with `tunnel: geneve` as suggested in cilium example values file. Initially it worked fine but later when I tried to deploy bigger helm chart, it started throwing `http2: client connection lost`.

After digging around, I've found it may an issue with Cilium's auto MTU which is setting MTU to 1500 from external interface but private interface has 1450 MTU. As per suggestions, I changed Cilium MTU to different values, 1300 seems to have worked for many, without any success.

relevant cilium issue [14339](https://github.com/cilium/cilium/issues/14339)

Later I've changed tunnel to `vxlan` and everything started working fine. No connection issues any more.

Anyone faced similar issue?

**Comment by mysticaltech** [comment, marked as answer]
Thanks for sharing @hariprasadiit. I've not used it enough to notice that issue. I will update the example to remove geneve, as the default is vxlan.

Don't forget the cilium-cli connectivity test, it seemed really thorough the last time I used it.

**Comment by hariprasadiit** [comment]
@mysticaltech Thanks for the suggestion. I'll try cilium-cli next time when I face network issues.
